design: Setup shadcn, upgrade storybook, story for Login page

nithindavid opened the pull request:
<img width="1194" height="622" alt="image" src="https://github.com/user-attachments/assets/25fe7415-9155-449d-9d4c-3bd30af714a2" />

nithindavid committed:
Update package dependencies, add new components, and enhance login functionality

- Added `shadcn` dependency to package.json and package-lock.json.
- Introduced new `LoginForm` component for user authentication.
- Updated styles and structure in `frontend/components.json` and `frontend/src/styles/globals.css`.
- Enhanced existing UI components and updated Radix UI dependencies.
- Refactored onboarding background wrapper to integrate the new login form.

nithindavid committed:
Update package dependencies, enhance Storybook configuration, and introduce new authentication components

- Updated Storybook addons and configurations to use the latest versions.
- Added `LoginForm` and `PasswordInput` components for user authentication.
- Refactored existing components and added new UI elements for improved user experience.
- Enhanced the onboarding background wrapper to integrate the new login form.
- Cleaned up and organized the package.json and package-lock.json files for better dependency management.

nithindavid committed:
Add GitHub and Google sign-in buttons, along with corresponding icons and stories

- Introduced `GitHubSigninButton` and `GoogleSigninButton` components for authentication.
- Added `GitHub` and `Google` icons to the solid icons collection.
- Updated the `SolidIcons` index to include new icons.
- Created Storybook stories for both sign-in buttons to demonstrate usage and variations.
- Refactored `LoginForm` to support external validation and improved input handling.

nithindavid committed:
Enhance LoginForm component and update dependencies

- Added `util` package to package.json and package-lock.json for utility functions.
- Updated `LoginForm` to include new props for organization name and signup visibility.
- Refactored LoginForm styles and structure for improved readability and functionality.
- Enhanced Storybook stories for LoginForm to support new props and scenarios.
- Updated webpack configuration to resolve the `util` module correctly.

nithindavid committed:
Add Forgot Password components and enhance authentication flow

- Introduced `ForgotPasswordForm` and `ForgotPasswordInfoScreen` components for improved password recovery functionality.
- Implemented form validation and state management for the `ForgotPasswordForm`.
- Created Storybook stories for both components to demonstrate various use cases and configurations.
- Updated `LoginForm` styles for consistency with new components.
- Enhanced overall user experience in the authentication flow.

adishM98 committed:
feat: add GitHub Actions workflow for rendering Storybook previews

adishM98 committed:
chore: add dummy PR comment to Storybook workflow

adishM98 committed:
chore: comment out environment variables in Storybook workflow

nithindavid committed:
Add SetupAdminForm component and corresponding Storybook stories

- Introduced `SetupAdminForm` component for admin account setup, featuring fields for name, email, and password with validation.
- Implemented customizable props for labels, placeholders, and button text to enhance usability.
- Created comprehensive Storybook stories to showcase various configurations and states of the `SetupAdminForm`, including validation and loading scenarios.
- Ensured accessibility and user-friendly design in the form layout.

adishM98 committed:
fix: update repository URL and remove commented environment variables in Storybook workflow

adishM98 committed:
fix: update repository URL and add environment variables for Node and NPM versions in Storybook workflow

adishM98 committed:
chore: update dummy PR comment in Storybook workflow

adishM98 committed:
Merge pull request #14088 from ToolJet/component-library-setup-dummy-1

chore: update dummy PR comment in Storybook workflow

adishM98 committed:
Merge pull request #14087 from ToolJet/component-library-setup-dummy

add PR comment to Storybook workflow

nithindavid committed:
Add SignupSuccessInfo component and corresponding Storybook stories

- Introduced `SignupSuccessInfo` component to display a success message after user signup, including options for email, name, and additional info.
- Implemented customizable props for header text, message content, resend button, and back button functionality.
- Created comprehensive Storybook stories to showcase various configurations and states of the `SignupSuccessInfo` component, enhancing usability and testing.
- Ensured accessibility and user-friendly design in the component layout.

adishM98 committed:
fix: update deployment URLs and names to use 'storybook' prefix in Storybook workflow

nithindavid committed:
Enhance form components with Label integration and Button updates

- Added `Label` components to `ForgotPasswordForm`, `LoginForm`, and `SetupAdminForm` for improved accessibility and user experience.
- Updated `Button` component to support Lucide icons with dynamic loading, enhancing visual consistency and flexibility.
- Refactored utility functions in `ButtonUtils` to accommodate new icon handling logic.
- Expanded Storybook stories for `Button` to demonstrate new Lucide icon capabilities and various button configurations.
- Improved `Input` component styling to support dynamic text sizes based on input size prop.

nithindavid committed:
Refactor button and input components for improved styling and consistency

- Updated button size variants in `Button.jsx` to ensure consistent padding and border-radius across sizes.
- Introduced `buttonSizeVariants` utility in `Input.jsx` for dynamic button sizing, enhancing reusability.
- Modified interactive overlay colors in `componentdesign.scss` to use RGBA for better transparency control and visual consistency.

nithindavid committed:
Add Alert component and related Storybook stories

- Introduced `Alert` component for displaying various types of alerts (info, warning, danger) with customizable styles and icons.
- Created comprehensive Storybook stories for the `Alert` component, showcasing different variants, including dark mode and custom icon sizes.
- Added `InlineInfo` component to provide a compound alert with title, description, and optional button, enhancing user feedback options.
- Updated `LoginForm` and `AuthLayout` components to integrate new alert functionalities, improving user experience during authentication processes.

nithindavid committed:
Refactor Auth component stories to use 'Auth/Blocks' naming convention

- Updated titles of various Auth component stories to follow the new 'Auth/Blocks' structure for better organization.
- Removed several unused story exports from `AuthCenteredLayout`, `AuthLayout`, `ForgotPasswordForm`, and `LoginPage` to streamline the codebase.
- Introduced a new `LoginFlow` story to demonstrate the complete login process, integrating the `LoginForm`, `GoogleSigninButton`, and `GitHubSigninButton` components.
- Enhanced the overall structure and readability of the Auth component stories in Storybook.

nithindavid committed:
Enhance Auth components with new stories and updates

- Updated `ForgotPasswordForm` to utilize the new `InlineInfoCompound` component for displaying admin contact information, improving visual consistency.
- Refactored `ForgotPasswordInfoScreen` story title to follow the 'Auth/Blocks' naming convention for better organization.
- Added new `ForgotPasswordFlow` and `ForgotPasswordInfoFlow` stories to showcase the complete forgot password process in Storybook.
- Cleaned up unused exports in `ForgotPasswordForm` stories to streamline the codebase.

nithindavid committed:
Add new authentication forms and update Storybook organization

- Introduced `InviteSignupForm`, `JoinWorkspaceForm`, and `SignInWithPrefilledEmail` components for enhanced user authentication flows.
- Updated Storybook stories to include new `TeamInviteFlow`, showcasing the signup and workspace joining processes.
- Refactored existing story titles to follow the 'Auth/Blocks' naming convention for improved organization and clarity.

nithindavid committed:
Add SignupFormWithSSO component and update related stories

- Introduced `SignupFormWithSSO` component for enhanced user signup experience with support for Single Sign-On (SSO) options.
- Updated `InviteSignupForm` and `JoinWorkspaceForm` components to improve header styling and layout consistency.
- Created new Storybook stories for `SignupFormWithSSO` and related signup flows, showcasing various configurations and states.
- Organized stories under 'Auth/Flows' and 'Auth/Blocks' for better clarity and structure in the Storybook.

nithindavid committed:
Update package-lock.json with dependency version changes and additions

- Downgraded versions for several Radix UI components and class-variance-authority for compatibility.
- Updated Storybook-related packages to version 7.2.1 for improved features and stability.
- Added new dependencies for Babel plugins and presets to enhance development capabilities.
- Removed deprecated dependencies and cleaned up unused entries to streamline the package structure.

devanshu052000 commented on frontend/src/components/ui/Input/Input.jsx:
```diff
@@ -1,11 +1,28 @@
 
-const Input = React.forwardRef(
-  ({ className, size, type, multiline, response, isWorkspaceConstant, rows = 3, ...props }, ref) => {
+const buttonSizeVariants = cva('', {
+  variants: {
+    size: {
+      small: 'tw-text-[12px]/[18px] tw-px-[10px] tw-py-[6px] tw-rounded-[6px]',
+      medium: 'tw-text-[12px]/[18px] tw-px-3 tw-py-[7px] tw-rounded-[6px]',
+      large: 'tw-text-[14px]/[20px] tw-px-3 tw-py-[7px] tw-rounded-[8px]',
+    },
+  },
+  defaultVariants: {
+    size: 'medium',
+  },
+});
```
What is the use of this? since we are adding buttonSizeVariants to the input instead of a button

devanshu052000 commented on frontend/src/components/ui/Input/Input.jsx:
```diff
@@ -23,12 +40,20 @@ const Input = React.forwardRef(
 
     const validationClass = response === true ? 'valid-textarea' : response === false ? 'invalid-textarea' : '';
 
+    const inputStyle = ` placeholder:tw-text-text-placeholder ${
+      isValid === true ? '!tw-border-border-success-strong' : isValid === false ? '!tw-border-border-danger-strong' : ''
+    }`;
+
```
Aren't we already passing this down from the wrapper components `NumberInput` and `TextInput` components? If we are adding it to the base component,  can we make sure we aren't passing down redundant classes from wrapper ones? Will make the code a lot cleaner I feel. What are your thoughts?

devanshu052000 commented on frontend/src/components/ui/Input/Input.jsx:
```diff
@@ -37,23 +62,34 @@ const Input = React.forwardRef(
-              <SolidIcon className="eye-icon" name="eye" />
-            ) : (
-              <SolidIcon className="eye-icon" name="eyedisable" />
+          <div className="tw-relative">
```
Can we revert this if not necessary? This might break the existing component being used inside the codebase.

nithindavid committed:
Merge branch 'lts-3.16' into design/component-library-setup

nithindavid committed:
feat: enhance input components

- Added `EncryptedField` component for handling encrypted inputs.
- Updated `InputComponent` to conditionally render `EncryptedField` for password types.
- Introduced `PasswordInput` for password handling with visibility toggle.
- Refactored `CommonInput` to support dynamic input types including password.
- Enhanced validation messaging and added new validation states in stories.
- Updated styles and utility functions for better input handling and icon positioning.

nithindavid commented:
@devanshu052000 Thanks for the detailed review, I have fully revamped the Input & Child components to fully utilise the cva to support all types of sizes and variants.

Also, extracted the logic for Encrypted field & workspace Constant [Needs to test]

nithindavid commented on frontend/src/components/ui/Input/Input.jsx:
```diff
@@ -37,23 +62,34 @@ const Input = React.forwardRef(
-            {...props}
-          />
-        )}
-        {isPasswordField && !multiline && (
```
Separated the password from Input, so it won't be multiline
